packages/firestore/firestore.rules

```diff
@@ -1,9 +1,30 @@
 rules_version = '2';
 service cloud.firestore {
   match /databases/{database}/documents {
-    match /{document=**} {
-      allow read, write: if
-          request.time < timestamp.date(2021, 1, 16);
+
+    match /users/{userId} {
+      allow read: if userId == authUserUid();
+      allow create: if userId == authUserUid() && isValidUserCreate();
     }
+
+    function isValidUserCreate() {
+      return newData().keys().hasOnly(['role', 'devOpsMaturity', 'email'])
+        && newData().role is string
+        && newData().devOpsMaturity is string
```

Comment:
You should be able to retrieve the document for the devOpsMaturities and check if devOpsMaturity is between them